lib/github.ml

```diff
@@ -266,14 +267,12 @@ let assert_tag_exists ~dry_run tag =
 
 let publish_distrib ~dry_run ~msg ~archive ~yes p =
   let git_for_repo r = Cmd.of_list (Cmd.to_list @@ Vcs.cmd r) in
-  let curl = Cmd.(v "curl") in
   ( match Pkg.distrib_user_and_repo p with
   | Error _ as e -> if dry_run then Ok (D.user, D.repo) else e
   | r -> r )
   >>= fun (user, repo) ->
   Pkg.tag p >>= fun tag ->
   assert_tag_exists ~dry_run tag >>= fun () ->
-  OS.Cmd.must_exist curl >>= fun curl ->
```

Comment:
Is there any particular reason why we removed this check?

I'm not sure how this was reported and how this is reported now but it miight be worth taking a look. What ultimately matters is that we end up with a good error message if `curl` isn't installed. Would mind checking how it was and how it is now?

Reply:
Previously we would get an `Error` here, with this PR, we get an `Error` when calling `curl_create_release` then `run_with_auth` (the Error is returned by `Curly.run`) so the only difference is App_log printed more stuff before getting the error.

Reply:
That's fine by me, do you know what is the error message like when curl is missing?

Reply:
On master the message was `main.exe: [ERROR] curl: no such command in PATH`, it became:
```
main.exe: [ERROR] curl execution failed: Non 0 exit code Exit code
                  127
                  {status=Exit code 127; stderr="/bin/sh: 1: curl: not
                  found
                  ";
                  stdout=""}
```
the message from Curly is not very clear indeed, I kept the check just before calling `Curly.run`